src/summary.cpp

```diff
@@ -0,0 +1,140 @@
+#include <lo2s/config.hpp>
+#include <lo2s/summary.hpp>
+#include <lo2s/util.hpp>
+
+#include <boost/filesystem.hpp>
+
+#include <fstream>
+#include <iomanip>
+#include <iostream>
+#include <numeric>
+#include <ratio>
+#include <array>
+
+extern "C" {
+#include <sys/time.h>
+}
+
+namespace lo2s
+{
+
+Summary& summary()
+{
+    static Summary s;
+    return s;
+}
+
+Summary::Summary()
+: start_wall_time_(std::chrono::steady_clock::now()), num_wakeups_(0),
+  thread_count_(0), trace_dir_(""), exit_code_(0)
+{
+}
+
+void Summary::register_process(pid_t pid)
+{
+    pids_.emplace(pid);
+}
+
+void Summary::record_perf_wakeups(std::size_t num_wakeups)
+{
+    num_wakeups_ += num_wakeups;
+}
+
+void Summary::set_exit_code(int exit_code)
+{
+    exit_code_ = exit_code;
+}
+
+void Summary::add_thread()
+{
+    thread_count_++;
+}
+
+std::string pretty_print_bytes(std::size_t trace_size)
+{
+    double result_size = trace_size;
+    int unit = 0;
+
+    std::ostringstream out;
+    std::array<std::string, 5> units = { "B", "KiB", "MiB", "GiB", "TiB" };
+
+    while(result_size > 1024)
+    {
+        result_size /= 1024;
+        unit++;
+
+        //We can not get higher than TiB so break here
+        if(unit == units.size() - 1)
+        {
+            break;
+        }
+    }
+
+    out << std::fixed << std::setprecision(2) << result_size << units[unit];
+    return out.str();
+}
+
+void Summary::set_trace_dir(const std::string& trace_dir)
+{
+    trace_dir_ = trace_dir;
+}
+
+void Summary::show()
+{
+    std::chrono::duration<double> wall_time, cpu_time;
+
+    std::size_t trace_size;
+
+    if (config().quiet)
+    {
+        return;
+    }
+
+    wall_time = std::chrono::steady_clock::now() - start_wall_time_;
+    cpu_time = get_cpu_time();
+
+    boost::filesystem::recursive_directory_iterator it(trace_dir_), end;
+
+    trace_size = std::accumulate(it, end, 0,
+                [](std::size_t sum, boost::filesystem::directory_entry& entry)
+                {
+                    if (!boost::filesystem::is_directory(entry))
+                    {
+                        return sum + boost::filesystem::file_size(entry);
+                    }
+                    return sum;
+                });
+
+
+    std::cout << "[ lo2s: ";
+
+    if (config().monitor_type == lo2s::MonitorType::PROCESS)
+    {
+        for (const auto command : config().command)
+        {
+            std::cout << command << ' ';
+        }
+
+        std::cout << " (" << exit_code_ << "), ";
+        std::cout << thread_count_ << " threads, ";
+    }
+    else
+    {
+        std::cout << "-a mode, ";
```

Comment:
Should we change `-a mode` to `system mode` or even `system monitoring`?
@tilsche